libbeat/metric/system/numcpu/cpu_linux.go

```diff
@@ -0,0 +1,93 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package numcpu
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/v7/libbeat/paths"
+)
+
+// getCPU implements NumCPU on linux
+// see https://www.kernel.org/doc/Documentation/admin-guide/cputopology.rst
+func getCPU() (int, bool, error) {
+
+	// These are the files that LSCPU looks for
+	// This will report online CPUs, which are are the logical CPUS
+	// that are currently online and scheduleable by the system.
+	// Some users may expect a "present" count, which reflects what
+	// CPUs are available to the OS, online or off.
+	// These two values will only differ in cases where CPU hotplugging is in affect.
+	// This env var swaps between them.
+	_, isPresent := os.LookupEnv("LINUX_CPU_COUNT_PRESENT")
```

Comment:
I am ok with having this environment variable by now to control this behaviour, specially if we don't expose it to users. But I wonder some things:
* Should this be documented? or this is something we don't want to expose to users, unless requested?
* Is it possible that other operating systems have also the difference between online and offline CPUs? If that's the case, should we use a name that is not linux specific?
* What do you think about exposing this toggle as a setting in the system module instead of using an environment variable?

Reply:
I'm not entirely sure. On a lot of systems, this will functionally be a no-op, as far as metrics go. Other OSes definitely have something like this, but for now it's linux-only.